jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/ExplorerToolWindow.kt

```diff
@@ -65,70 +71,67 @@ class ExplorerToolWindow(private val project: Project) : SimpleToolWindowPanel(t
 
         project.messageBus.connect().subscribe(ProjectAccountSettingsManager.ACCOUNT_SETTINGS_CHANGED, this)
 
-        updateModel()
+        treePanelWrapper.setContent(errorPanel)
     }
 
     override fun settingsChanged(event: AccountSettingsEvent) {
-        updateModel()
-    }
-
-    internal fun updateModel() {
-        if (!projectAccountSettingsManager.hasActiveCredentials()) {
-            treePanelWrapper.setContent(errorPanel)
-            return
-        }
-
-        val model = DefaultTreeModel(DefaultMutableTreeNode())
-        val newTree = createTree(model)
-        val builder = AwsExplorerTreeBuilder(newTree, model, project)
-        Disposer.register(project, builder)
-        treePanelWrapper.setContent(ScrollPaneFactory.createScrollPane(newTree))
-
-        awsTree?.let {
-            AbstractTreeBuilder.getBuilderFor(it)?.let {
-                Disposer.dispose(it)
+        if (!event.isLoading) {
+            runInEdt {
+                if (!projectAccountSettingsManager.hasActiveCredentials()) {
+                    treePanelWrapper.setContent(errorPanel)
+                } else {
+                    refreshTree()
+                    treePanelWrapper.setContent(awsTreePanel)
+                }
             }
         }
+    }
 
-        awsTree = newTree
+    internal fun refreshTree() {
```

Comment:
Consider renaming to `invalidateTree` to match the underlying method?

Reply:
Done